### ISSUE TITLE
descriptor_management sample gives misleading results

### ISSUE THREAD

**Issue by DmitryTsyganov** [KhronosGroup/Vulkan-Samples#527]
### Premise
In the README file of descriptor_management sample, it is stated that the sample shows how slow _"simplest descriptor management scheme"_ is and that introducing a descriptor set cache used in the framework _"has a dramatic effect on frame time"_. The sample itself allows setting **Descriptor set caching** to **Enabled** or **Disabled**. One might conclude that this sample demonstrates the performance gain that can be achieved by introducing a descriptor set cache into a Vulkan app.

### Problem
By looking into VS profiler when the cache is disabled and then looking in the code I noticed that all choosing the **Disabled** option
does is it clears descriptors set cache and resets pools for the current frame. The cache is still queried and filled.
The sample doesn't show the performance difference between recreating descriptor sets every frame and using descriptor set cache.
It shows the difference between 
getting descriptor sets from filled descriptor cache
and 
querying empty cache, recreating descriptor sets, putting them into the cache, and clearing the cache.
Obviously, the first option is always faster, it's the same, but less work.

Simply put, **Descriptor set caching** is never truly **Disabled** in this sample.

### Motivation
Implementing descriptor set cache in an engine/renderer is a tradeoff between saving time on creating cached objects (here: descriptor sets) and introducing cache overhead (hashing, comparing, copying, possibly allocating memory).
